addons/spreadsheet_dashboard/static/src/bundle/dashboard_action/dashboard_action.js

```diff
@@ -80,6 +80,12 @@ export class SpreadsheetDashboardAction extends Component {
                 };
             },
         });
+
+        onWillDestroy(() => {
+            for (const dashboard of Object.values(this.loader.dashboards)) {
+                this.loader.stopListeningToDataSourceUpdate(dashboard);
+            }
+        });
```

Comment:
I don't understand how it leaks here.
If the component is destroyed, nothing retains a reference to it. Which means nothing has a reference to `this.loader` and hence nothing has any reference to `model` nor to `odooDataProvider`, which means they both can be garbage collected.

Of course, the first assumption *"If the component is destroyed, nothing retains a reference to it"* needs to be true... and it's not because of the leak in the action manager, right ?

It'd be interesting to see if this is really a leak when the one of the action manager is fixed